File: Нейронные сети/lpc_nn/source/lpc55-C/preemp.c
/********************************************************************
*
*	PREEMP Version 48
*
********************************************************************
*
*   Preemphasize speech with      ( 1 - .9375z**-1 )  [old preemphasis]
*   cascaded with ( 1 + .2z**-1 ) / ( 1 + .5z**-1 )   [6 db/oct ramp]
*
* Inputs:
*  NSAMP  - Number of samples to filter
*  INBUF  - Input speech buffer
*  COEF	  - Preemphasis coeficient
* Output:
*  PEBUF  - Preemphasized speech buffer (can be equal to INBUF)
*  Z      - Filter state
*/

preemp( inbuf, pebuf, nsamp, coef, z )
int nsamp;
float *inbuf, *pebuf, coef, *z;
{
int i;
float temp=0.0;


for(i=1; i<=nsamp; i++) {
	temp = inbuf[i] - *z + coef**(z+1);
	*(z+1) = *z;
	*z = inbuf[i];
	pebuf[i] = temp;
}


}

File: Нейронные сети/lpc_nn/source/lpc55/preemp.f
********************************************************************
*
*	PREEMP Version 55
*
********************************************************************
*
*   Preemphasize speech with a single-zero filter.
*  (When coef = .9375, preemphasis is as in LPC43.)
*
* Inputs:
*  NSAMP  - Number of samples to filter
*  INBUF  - Input speech buffer
*  COEF   - Preemphasis coefficient
* In/Out:
*  Z      - Filter state
* Output:
*  PEBUF  - Preemphasized speech buffer (can be equal to INBUF)
*
	subroutine preemp(inbuf, pebuf, nsamp, coef, z)
	integer nsamp, i
	real temp, inbuf(nsamp), pebuf(nsamp), coef, z

	do 10 i = 1, nsamp
	    temp = inbuf(i) - coef*z
	    z = inbuf(i)
	    pebuf(i) = temp
10	continue

	return
	end
